unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/LOD.cs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        private Renderer CreateImpostor()
+        {
+            GameObject quadImpostorContainer = Object.Instantiate(Resources.Load<GameObject>("QuadImpostor"), impostorContainer.transform);
+            return quadImpostorContainer.GetComponent<Renderer>();
+        }
+
+        private async UniTaskVoid Transition(CancellationToken ct)
+        {
+            if (ct.IsCancellationRequested)
+                throw new OperationCanceledException();
+
+            try
+            {
+                visibility.SetCombinedRendererVisibility(true);
+                impostorRenderer.enabled = true;
+                float targetAvatarAlpha = lodIndex <= 1 ? 1f : 0f;
+                while (!Mathf.Approximately(targetAvatarAlpha, avatarAlpha))
+                {
+                    avatarAlpha = Mathf.MoveTowards(avatarAlpha, targetAvatarAlpha, (1f / TRANSITION_DURATION) * Time.deltaTime);
+                    UpdateAlpha(avatarAlpha);
+                    await UniTask.NextFrame(ct);
+                }
+
+                UpdateSSAO(lodIndex);
+                UpdateMovementLerping(lodIndex);
+
+                visibility.SetCombinedRendererVisibility(avatarAlpha > 0);
+                visibility.SetFacialFeaturesVisibility(lodIndex == 0);
+
+                impostorRenderer.enabled = avatarAlpha == 0;
+            }
+            catch (OperationCanceledException)
+            {
+                //No disposing required
+                throw;
+            }
+        }
+
+        private void UpdateAlpha(float avatarAlpha)
+        {
+            // Validate that sharedMaterials must be used here
+            Material[] mats = combinedAvatar.sharedMaterials;
+            for (int j = 0; j < mats.Length; j++)
+            {
+                mats[j].SetFloat(AvatarSystemUtils.DitherFade, avatarAlpha);
+            }
+
+            Material impostorMaterial = impostorRenderer.material;
+            //TODO implement dither in Unlit shader
+            Color current = impostorMaterial.GetColor(AvatarSystemUtils._BaseColor);
+            current.a = 1f - avatarAlpha;
+            impostorMaterial.SetColor(AvatarSystemUtils._BaseColor, current);
+        }
+
+        private void UpdateSSAO(int lodIndex)
+        {
+            Material[] mats = combinedAvatar.sharedMaterials;
+
+            for (int j = 0; j < mats.Length; j++)
+            {
+                if (lodIndex == 0)
+                    mats[j].DisableKeyword(AvatarSystemUtils.SSAO_OFF_KEYWORD);
+                else
+                    mats[j].EnableKeyword(AvatarSystemUtils.SSAO_OFF_KEYWORD);
+            }
+        }
+
+        private void UpdateMovementLerping(int lodIndex) { avatarMovementController.SetMovementLerpWait(lodIndex >= 2 ? AvatarRendererHelpers.IMPOSTOR_MOVEMENT_INTERPOLATION : 0f); }
+
+        public void Dispose()
+        {
+            transitionCTS?.Cancel();
+            transitionCTS = new CancellationTokenSource();
```

Comment:
transitionCTS is missing Dispose call. Also, its kind of confusing to generate a new instance in a disposal method. In case it needs to be reused, you could create it in Transition method when is null or already disposed